helix-core/src/test/java/org/apache/helix/manager/zk/TestZkHelixAdmin.java

```diff
@@ -560,6 +560,93 @@ public void testLegacyEnableDisablePartition() {
         2);
   }
 
+  @Test
+  public void testResetPartition() {
+    String className = TestHelper.getTestClassName();
+    String methodName = TestHelper.getTestMethodName();
+    String clusterName = className + "_" + methodName;
+    String instanceName = "TestInstance";
+    String testResource = "TestResource";
+    System.out.println("START " + clusterName + " at " + new Date(System.currentTimeMillis()));
+    HelixAdmin admin = new ZKHelixAdmin(_gZkClient);
+    admin.addCluster(clusterName, true);
+    admin.addInstance(clusterName, new InstanceConfig(instanceName));
+    admin.enableInstance(clusterName, instanceName, true);
+
+    InstanceConfig instanceConfig = admin.getInstanceConfig(clusterName, instanceName);
+
+    // Test the sanity check for resetPartition.
+    // resetPartition is expected to throw an exception when provided with a non-exist instance.
+    try {
+      admin.resetPartition(clusterName, "WrongTestInstance", testResource,
+          Arrays.asList("1", "2"));
+      Assert.fail("Should throw HelixException");
+    } catch (HelixException expected) {
+      // This exception is expected because the instance name is made up.
+      Assert.assertEquals(expected.getMessage(),
+          "Can't reset state for " + testResource  + "/[1, 2] on WrongTestInstance," +
+              " because WrongTestInstance never existed in cluster " + clusterName);
+    }
+
+    // resetPartition is expected to throw an exception when provided with a non-alive instance.
+    try {
+      admin.resetPartition(clusterName, instanceName, testResource,
+          Arrays.asList("1", "2"));
+      Assert.fail("Should throw HelixException");
+    } catch (HelixException expected) {
+      // This exception is expected because the instance is not alive.
+      Assert.assertEquals(expected.getMessage(),
+          "Can't reset state for " + testResource  + "/[1, 2] on " + instanceName
+              + ", because " + instanceName + " is not alive in cluster " + clusterName + " anymore");
+    }
+
+    HelixManager manager = initializeHelixManager(clusterName, instanceConfig.getInstanceName());
+    try {
+      manager.connect();
+    } catch (Exception e) {
```

Comment:
No need to catch this exception. If an exception is thrown from connect(), the test will throw this exception and fail.

Reply:
I think I need to catch it so that compile won't fail.